test/amqp/client_test.rb

```diff
@@ -211,6 +211,83 @@ def test_it_can_select_confirm
     assert channel.wait_for_confirm id
   end
 
+  def test_it_can_qos
+    client = AMQP::Client.new("amqp://localhost")
+    connection = client.connect
+    channel = connection.channel
+    q = channel.queue_declare ""
+    channel.basic_qos 0, 1
+    channel.basic_publish "foo", "", q[:queue_name]
+    channel.basic_publish "bar", "", q[:queue_name]
+    i = 0
+    channel.basic_consume(q[:queue_name], no_ack: false) do |msg|
+      channel.basic_cancel msg.consumer_tag
+      i += 1
```

Comment:
This will only be called once, because you've canceled the consumer on the line before